textdb/textdb-web/src/main/java/edu/uci/ics/textdb/web/request/beans/RegexSourceBean.java

```diff
@@ -65,18 +65,19 @@ public void setDataSource(String dataSource) {
     public boolean equals(Object other) {
         if (other == null) return false;
         if (other == this) return true;
-        if (!(other instanceof OperatorBean)) return false;
+        if (!(other instanceof RegexSourceBean)) return false;
         RegexSourceBean regexSourceBean = (RegexSourceBean) other;
         return new EqualsBuilder()
+                .appendSuper(super.equals(regexSourceBean))
                 .append(regex, regexSourceBean.getRegex())
                 .append(dataSource, regexSourceBean.getDataSource())
-                .isEquals() &&
-                super.equals(regexSourceBean);
+                .isEquals();
     }
 
     @Override
     public int hashCode() {
         return new HashCodeBuilder(17, 31)
+                .append(super.hashCode())
```

Comment:
This is not required because, we call the `equals()` function on the super class too, which employs the super's `hashCode()` function.